### PR TITLE
Intro to c++ lesson

### PR DESCRIPTION
**Pull request by AlannaFey**
Still need to add some problems, but this is what I currently have for the lessons
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: public/lessons/intro-to-cpp/intro.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+The first thing you should do when writing code in C++ is make sure that you have a template to work with.
+This is the basic structure that your template should follow:
+
+```c++
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+    ios_base::sync_with_stdio(false);
+    cin.tie(NULL);
```
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/intro.mdx:21]: I'd prefer teaching `cin.tie(0)->sync_with_stdio(0)` instead.

File: public/lessons/intro-to-cpp/intro.mdx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Why C++?
+
+Many of the examples you will find on O(fast) are written in C++.  This is for a few reasons.
+C++ is a fast language since it has access to many low-level operations.
+It also has a wide variety of built-in data structures useful in competitive programming.
```
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/intro.mdx:5]: I'd mention it as the "standard library", which also has built-in algorithms.

File: src/lessons.json [hunk 3]
```diff
@@ -37,5 +37,16 @@
...
     "lessonGroup": "Intro Data Structures",
     "files": []
+  },
+  "intro-to-cpp": {
+    "lessonName": "Intro to C++",
+    "lessonGroup": "Intro to Competitive Programming",
+    "files": [
+      "intro.mdx",
+      "input-output.mdx",
+      "io-fitbq.mdx",
```
Comment (AsirAAlam) [inline src/lessons.json:47]: super nit: can we follow existing convention and say `io-fitb.mdx` instead? (without the q)

File: public/lessons/intro-to-cpp/io-fitbq.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Input/Output Practice
+<FITBBlock hint="Think about the way that the variables go into `cin` and `cout`">
+    Lets say that you want to take in a variable `x` and print it out in C++. Fill in the correct arrows to do this.
```
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/io-fitbq.mdx:3]: Let's

Comment (AsirAAlam) [reply public/lessons/intro-to-cpp/io-fitbq.mdx:3]: ~~take in a variable x~~ -> input to a variable x

File: public/lessons/intro-to-cpp/source/ascii.png
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/source/ascii.png:1]: remove? doesn't look like you're using this

File: public/lessons/intro-to-cpp/intro.mdx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+You also get the choice to manage memory yourself, similar to C, but without a lot of the nuance that comes with programming in C.
+Overall, we believe that coding in C++ is beneficial in your journey of competitive programming.
+If you don’t know much about C++, this section will help you learn the basics.
```
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/intro.mdx:8]: If you are new to C++

**Comment by AsirAAlam** [comment]
This lesson in particular could use quite a few links to resources or C++ docs. Could even add a small section recommending the use of docs or the internet in general to learn more about C++.

File: public/lessons/intro-to-cpp/intro.mdx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+We `return 0;` at the end of the program because it is the standard way of exiting the main function.  If you choose to
+return a non-zero value, you may run into runtime errors on many judges.
+
+
+# Compiling and Running
+
+C++ code needs to be compiled before running.  To do this you should use `g++ hello.cpp` at the command line where `hello.cpp` is the name of your file.
+To run the program on Linux or Mac use `./a.out`.  If you are using Windows, we recommend using [Windows Subsystem for Linux](https://ubuntu.com/desktop/wsl).
+You can also name your program something else using `-o` like so: `g++ hello.cpp -o hello`.  Then to run this program you run `./hello`.
```
Comment (AsirAAlam) [inline public/lessons/intro-to-cpp/intro.mdx:45]: should, more importantly, include some essential compile flags like `-O2` and `-std=c++17`

**Comment by gblake31** [comment]
I can hear your voice in the text, which I think is great. I feel like you're teaching it to me. The content is really concise, like you're giving me a lot of information in a small amount of text. Great use of code snippets, tables, and fitb!

**Comment by gblake31** [comment]
I wrote a problem for HSPT online '22-23 that requires only IO and a test case loop. (Sequel) 
We could consider using that one. Although, maybe this lesson doesn't need any problems. Up to you.

**Review by gblake31 — APPROVED** [review]

**Pull request merged**